doc/lua_api.md

```diff
@@ -8655,6 +8657,18 @@ Used by `minetest.register_node`, `minetest.register_craftitem`, and
     liquids_pointable = false,
     -- If true, item can point to all liquid nodes (`liquidtype ~= "none"`),
     -- even those for which `pointable = false`
+    
+    pointabilities = {
+		["leaves"] = false,
+		["crumbly"] = "blocking"
```

Comment:
might be better to change this to allow item names or group names as `group:name_here`
we have something similar in another place and it would not force you to create a group if you only want to affect one item.

Reply:
At first I implemented it this way with `group:name_here` and `modname:name`, but changed it to just groups.
The main reason why I did this was because it does not distinguish node names and object names.
Also it seems like the neither the `ServerActiveObject` nor the child class `UnitSAO` stores the object's name.
Maybe would be sufficient to not accept object names and only object groups, but this would then be inconsistent, since it accepts node names.

I also though about adding `drawtype:name` to properly replace `liquids_pointable`, but in the end it just makes things unnecessary complicated.

Moreover, having to add a new group if you want to affect only one item, is not that bad in my opinion.
For digging time groups you have to do the same thing.

Reply:
it affects objects? well that's not clear from the documentation at all.

also, armor groups are for armor. no matter how convenient we shouldn't be misusing them for this feature.

> Also it seems like the neither the ServerActiveObject nor the child class UnitSAO stores the object's name

they definitely do
did you maybe mean the client classes? the client indeed doesn't know which entity it is on the server-side

Reply:
> > Also it seems like the neither the ServerActiveObject nor the child class UnitSAO stores the object's name
>
> they definitely do did you maybe mean the client classes?

Never mind, I just found the name in the `LuaEntitySAO` class which is a child of `UnitSAO`.
(the whole object code is a bit confusing, since it is not well documented and very different in the client and server implementation)


> the client indeed doesn't know which entity it is on the server-side

Are you sure about this?

The `GenericCAO` (which is a client Object) does have a `m_name` member variable.
It is also where I get the armor groups from. (Since for some reason they are not part of the `ObjectProperties`)
Also, the object name (unlike the armor groups and properties) can't be changed after initialization.

> also, armor groups are for armor. no matter how convenient we shouldn't be misusing them for this feature.

Well armor groups are the only groups that objects have and I have seen them to be used for all kinds of purpose in games, but you are right this shouldn't be the case for the engine.
Also, I thought, since node groups are not only for digging times, armor groups are not only for damage calculation, but this turns out to be wrong

So how should the property look like in the API?
Would this be an agreeable solution: 

- Separate lists for `nodes` (including `groups`) and `objects`
- A field for objects that are `players`
- A Default `fallback` that is chosen if none of the others match

Like this:

```
pointabilities = {
    nodes = {
        ["default:stone"] = "blocking", 
        ["group:leaves"] = false
    },
    objects = {
        ["modname:entityname"] = "true"
    },
   players = true,
   fallback = false
}
```
- Or a separate one for `groups`?
It may be a little faster to check and more readable in lua since you don't have care about the colon.



> it affects objects? well that's not clear from the documentation at all.

I did even document it at the `pointable` property of nodes and objects.
Maybe `A list of groups overriding the 'pointable' property while holding the item.` is unclear.

When the armor groups problem is resolved I will change it to something like
 `A list of groups overriding the 'pointable' property of pointed node and object.`

Reply:
Armor groups are already used for other things than armor, see "`ObjectRef` armor groups" in doc.
It's therefore imo ok to use them for this. But idk, maybe sfan5 knows a reason why they're inappropriate.

It might even be nice if armor groups were renamed to just `groups` / "object groups", and merged into obj properties. But that's out of scope for this PR.

Specifying pointabilities separately for nodes and objects makes sense, as the names don't live in the same namespace, AFAIK.
The `group:...` syntax is also good, for consistency with other parts of the api.
This seems fine:
```lua
pointabilities = {
    nodes = {
        ["default:stone"] = "blocking", 
        ["group:leaves"] = false,
    },
    objects = {
        ["modname:entityname"] = true, -- not required, if it's too hard to get the entity name to the client
        ["group:ghosty"] = true, -- (an armor group)
    },
    -- (no `player = ...` required, players have armor groups)
}
```

Reply:
Thank you for the response, I have changed the API accordingly.
Now everything should be fine, so I consider this as resolved.

Also I have changed that the entity name (and not just an empty string) gets sent to the client on initialization.
(I searched in the git history, but haven't found a reason why this has not been the case in the past.)
https://github.com/minetest/minetest/blob/a17b240a2781c64c53e11f089f3f43b75bb22d6c/src/server/luaentity_sao.cpp#L248